File: app/src/main/java/com/cere/csplayer/data/SharedPre.java
package com.cere.csplayer.data;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by CheRevir on 2019/10/22
 */
public class SharedPre {
    private static final String NAME = "share_preference";

    public static SharedPreferences getSharedPreferences(Context context) {
        return context.getApplicationContext().getSharedPreferences(NAME, Context.MODE_MULTI_PROCESS);
    }

    public static void putInt(Context context, String key, int value) {
        getSharedPreferences(context).edit().putInt(key, value).commit();
    }

    public static int getInt(Context context, String key, int defValue) {
        return getSharedPreferences(context).getInt(key, defValue);
    }

    public static void putString(Context context, String key, String value) {
        getSharedPreferences(context).edit().putString(key, value).commit();
    }

    public static String getString(Context context, String key, String defValue) {
        return getSharedPreferences(context).getString(key, defValue);
    }

    public static void putBoolean(Context context, String key, boolean value) {
        getSharedPreferences(context).edit().putBoolean(key, value).commit();
    }

    public static boolean getBoolean(Context context, String key, boolean defValue) {
        return getSharedPreferences(context).getBoolean(key, defValue);
    }
}

File: app/src/main/java/com/cere/csplayer/data/SharedPre.kt
package com.cere.csplayer.data

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences

/**
 * Created by CheRevir on 2019/10/22
 */
object SharedPre {
    private const val NAME = "share_preference"

    @Suppress("DEPRECATION")
    private fun getSharedPreferences(context: Context): SharedPreferences {
        return context.applicationContext.getSharedPreferences(NAME, Context.MODE_MULTI_PROCESS)
    }

    @SuppressLint("ApplySharedPref")
    @JvmStatic
    fun putInt(context: Context, key: String?, value: Int) {
        getSharedPreferences(context).edit().putInt(key, value).commit()
    }

    @JvmStatic
    fun getInt(context: Context, key: String?, defValue: Int): Int {
        return getSharedPreferences(context).getInt(key, defValue)
    }

    @SuppressLint("ApplySharedPref")
    fun putString(context: Context, key: String?, value: String?) {
        getSharedPreferences(context).edit().putString(key, value).commit()
    }

    fun getString(context: Context, key: String?, defValue: String?): String? {
        return getSharedPreferences(context).getString(key, defValue)
    }

    @SuppressLint("ApplySharedPref")
    @JvmStatic
    fun putBoolean(context: Context, key: String?, value: Boolean) {
        getSharedPreferences(context).edit().putBoolean(key, value).commit()
    }

    @JvmStatic
    fun getBoolean(context: Context, key: String?, defValue: Boolean): Boolean {
        return getSharedPreferences(context).getBoolean(key, defValue)
    }
}